src/webui/webapplication.cpp

```diff
@@ -858,12 +858,8 @@ void WebApplication::action_command_setCategory()
 
     foreach (const QString &hash, hashes) {
         BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
-        if (torrent) {
-            if (!torrent->setCategory(category)) {
-                status(400, "Incorrect category name");
-                return;
-            }
-        }
+        if (torrent)
+            torrent->setCategory(category);
```

Comment:
not sure if webAPI users will miss this, but you can easily preserve the old behavior by adding a `if (!findCategory(category)) return status(...);` check right?

Reply:
Before `TorrentHandle::setCategory()` could fail in case of incorrect category name. Now it's never fails. So this code is correct.

Reply:
OK, thanks for clarifying.